### DISCUSSION TITLE
Retriveal function doesn't work correctly

### DISCUSSION THREAD

**Discussion by comunicativi-admin** [danny-avila/LibreChat#3709] [Troubleshooting] [answered]
### What happened?

Problem 1
After adding multiple files to the assistant knowledge from the Assistant builder pannel on the right, when asking the assistant to retrieve the content of a specific file by name, the assistant seems to retrieve a random file.
This problem only occurs in the LibreChat interface, not when using the OpenAI playground.

Problem 2
When adding files to the assistant file search fron the OpenAI interface, the assistant is unable to find the file.
This problem only occurs in the LibreChat interface, not when using the OpenAI playground.

### Steps to Reproduce

1. Upload multiple files.
2. Ask the assistant for the content of one of the files.

### Screenshots

![Schermata 2024-08-20 alle 10 45 14](https://github.com/user-attachments/assets/30e70aff-a3dc-4edd-98a7-03fe9191589f)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Uploaded files via LibreChat are for Code interpreter only. This is because OpenAI uses “vector stores” now for assistants. This is expected behavior with the current implementation.

If you want to use files, upload them into your chat input instead or configure a vector store for the assistant from the OpenAI playground, then use it in LibreChat

**Comment by comunicativi-admin** [comment]
Thanks for the fast response.

In my usecase, the files should be accessible from every chat of every user, so uploading them in the chat is not an option.

As said in the bug report, if I upload files in a vector store from the OpenAI playground, the file retriveal work fine in the OpeanAI playground, but does not work on LibreChat.

I feel like I'm missing something.

![Schermata 2024-08-20 alle 14 08 06](https://github.com/user-attachments/assets/cc026226-3c7b-4a5f-8f29-0f9790a01a4c)

![Schermata 2024-08-20 alle 14 08 18](https://github.com/user-attachments/assets/99c93a93-185b-47c9-9374-b6cb479382fc)

**Reply by danny-avila** [reply, marked as answer]
You're using an old version of LibreChat, you should see "File Search" and not "Retrieval"
![image](https://github.com/user-attachments/assets/be027d71-8e60-418d-92d4-b6d09cc77c89)

Otherwise, use Assistants V1 from the OpenAI playground:

![image](https://github.com/user-attachments/assets/01bf6a0e-23f1-413b-a93f-12d9a8fba2f1)

**Reply by comunicativi-admin** [reply]
Thanks! I tested with the latest version and it works fine.

On my project I've done some heavy customizations and it will be a pain to switch to another version, but i see no alternative. 
Do you have any advice on how I could make changes to the repo and still beeing able to pull updates without doing a huge work?
